### PR TITLE
Feature: Add /schedule/cole-demand and /schedule/cole-demand/:id endpoints

### LINKED ISSUE #482: Add /schedule/cole-demand and /schedule/cole-demand/:id routes.
## User stories
As a user I want to create schedules with different schedules on different routes to keep creating schedules intuitive.
Therefore, please add /schedule/cole-demand and /schedule/cole-demand/:id routes.

## Acceptance Criteria
- [ ] Add validation
- [ ] Add /schedule/cole-demand route
- [ ] Add  /schedule/cole-demand/:id route

### PR DESCRIPTION
**Pull request by antonykamp**
Fixes #482 

Add endpoint and test for cole demand endpoints.

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: tests/api/test_api_demand_schedule.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                "train_schedule_train_type": "passenger",
+                "platforms": ["platform1", "platform2"],
+            },
+        ],
+    )
+    def test_post(self, client, clear_token, data):
+        response = client.post(
+            "/schedule/cole-demand", headers={TOKEN_HEADER: clear_token}, json=data
+        )
+        assert response.status_code != 422
```
Comment (CR1337) [inline tests/api/test_api_demand_schedule.py:45]: Check for the right code not one wrong code :)

File: tests/api/test_api_demand_schedule.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            "/schedule/cole-demand", headers={TOKEN_HEADER: clear_token}, json=data
+        )
+        assert response.status_code == 422
+
+    def test_get_single(self, client, clear_token):
+        object_id = uuid.uuid4()
+        response = client.get(
+            f"/schedule/cole-demand/{object_id}", headers={TOKEN_HEADER: clear_token}
+        )
+        assert response.status_code == 404
```
Comment (CR1337) [inline tests/api/test_api_demand_schedule.py:59]: Check for an existing one too please.

File: tests/api/test_api_demand_schedule.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    def test_delete(self, client, clear_token):
+        object_id = uuid.uuid4()
+        response = client.delete(
+            f"/schedule/cole-demand/{object_id}", headers={TOKEN_HEADER: clear_token}
+        )
+        assert response.status_code == 404
```
Comment (CR1337) [inline tests/api/test_api_demand_schedule.py:66]: Check for an existing one too please.

**Review by CR1337 — CHANGES_REQUESTED** [review]
Same as #501 and #502 :D

**Review by CR1337 — CHANGES_REQUESTED** [review]
Same as #501 and #502 :D

File: src/api/schedule.py [hunk 4]
```diff
@@ -1,3 +1,55 @@
...
+from webargs.flaskparser import parser
+
+from src import implementor as impl
+from src.api import schemas
+from src.api.decorators import token_required
 
 bp = Blueprint("schedule", __name__)
+
+
+@bp.route("/schedule/cole-demand", methods=["get"])
```
Comment (instant-sky) [inline src/api/schedule.py]: Shouldn't that be coal instead of cole everywhere?

**Review by instant-sky — COMMENTED** [review]
I think it should be coal instead of cole or am I missing something? Otherwise it's awesome to see all the progress with the api! 😄

File: tests/api/test_api_demand_schedule.py (hunk 1 above)
Comment (antonykamp) [reply tests/api/test_api_demand_schedule.py:45]: The idea behind the `test_api_*` tests was to verify that the routes exist and validation works. The `test_impl_*` tests verify the functionality. That's the reason why I only check if the return codes don't equal 422.

File: src/api/schedule.py (hunk 4 above)
Comment (antonykamp) [reply src/api/schedule.py]: Thanks for pointing out! Upsi! 😮

File: tests/api/test_api_demand_schedule.py (hunk 2 above)
Comment (antonykamp) [reply tests/api/test_api_demand_schedule.py:59]: You're right. I should test more things here :/ But it's consistent now with the rest of the `test_impl_*`tests. I suggest that I open an issue that addresses an improvement of the `test_impl_*` tests by mocking the implementor and verifying that it is called with the correct arguments and the status_code is correct.

Comment (antonykamp) [reply tests/api/test_api_demand_schedule.py:59]: https://github.com/BP2022-AP1/bp2022-ap1/issues/516

**Review by instant-sky — APPROVED** [review]
Awesome, thank you! 🚂⛏️

File: tests/api/test_api_demand_schedule.py (hunk 1 above)
Comment (CR1337) [reply tests/api/test_api_demand_schedule.py:45]: Ok, then go for it.

File: tests/api/test_api_demand_schedule.py (hunk 2 above)
Comment (CR1337) [reply tests/api/test_api_demand_schedule.py:59]: Ok, sounds good :)

**Review by CR1337 — APPROVED** [review]

**Pull request merged**